src/tests/end-to-end/tests/details-view/tabstop.test.ts

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { launchBrowser } from '../../common/browser-factory';
+import { fastPassSelectors } from '../../common/element-identifiers/common-selectors';
+import { TargetPageElementSelectors } from '../../common/element-identifiers/target-page-selectors';
+import { Page } from '../../common/page-controllers/page';
+import { PopupPage } from '../../common/page-controllers/popup-page';
+import { TargetPage } from '../../common/page-controllers/target-page';
+import { Browser } from './../../common/browser';
+
+describe('Tabstop tests', () => {
+    describe('tabstop from fastpass', () => {
+        let browser: Browser;
+        let targetPage: TargetPage;
+        let detailsViewpage: Page;
+
+        beforeAll(async () => {
+            browser = await launchBrowser({ suppressFirstTimeDialog: true });
+            targetPage = await browser.newTargetPage();
+            detailsViewpage = await browser.newDetailsViewPage(targetPage);
+
+            await goToTabStopTest();
+            await enableToggleByAriaLabel();
+        });
+
+        afterAll(async () => {
+            if (browser) {
+                await browser.close();
+            }
+        });
+
+        test('if visualHelper is turned on after starting tabstop and pressing tabs on target page', async () => {
+            await targetPage.bringToFront();
+            await targetPage.keyPress('Tab');
+
+            const nativeWidgetHeadingTargetPage = await targetPage.getSelectorElement(TargetPageElementSelectors.targetPageNativeWidgetH1);
+            await nativeWidgetHeadingTargetPage.click();
+
+            const frame = targetPage.frames().find(f => f.name() === TargetPageElementSelectors.targetPageNativeWidgetIFrameName);
+            const button = await frame.$(TargetPageElementSelectors.targetPageNativeWidgetFirstRadio);
+            await button.click();
+
+            const shadowRoot = await targetPage.getShadowRoot();
+            await targetPage.waitForDescendentSelector(shadowRoot, fastPassSelectors.tabStopVisulizationStart, { visible: true });
+
+            for (let i = 0; i < 3; i++) {
+                await targetPage.keyPress('Tab');
+            }
+
+            expect(await targetPage.getShadowRootHtmlSnapshot()).toMatchSnapshot();
+        });
+
+        async function goToTabStopTest(): Promise<void> {
```

Comment:
I think these 2 helpers (maybe slightly more general versions of them) would probably be more appropriate on the `DetailsViewPage`